rclcpp_lifecycle/test/test_lifecycle_node.cpp

```diff
@@ -492,31 +492,51 @@ TEST_F(TestDefaultStateMachine, test_getters) {
 TEST_F(TestDefaultStateMachine, test_graph) {
   auto test_node = std::make_shared<EmptyLifecycleNode>("testnode");
   auto names = test_node->get_node_names();
-  EXPECT_EQ(names.size(), 1u);
-  EXPECT_STREQ(names[0].c_str(), "/testnode");
+
+  EXPECT_NE(names.end(), std::find(names.begin(), names.end(), std::string("/testnode")));
 
   // parameter_events, rosout, /testnode/transition_event
   auto topic_names_and_types = test_node->get_topic_names_and_types();
-  EXPECT_EQ(topic_names_and_types.size(), 3u);
+  EXPECT_NE(
+    topic_names_and_types.end(),
+    topic_names_and_types.find(std::string("/testnode/transition_event")));
   EXPECT_STREQ(
     topic_names_and_types["/testnode/transition_event"][0].c_str(),
     "lifecycle_msgs/msg/TransitionEvent");
 
   auto service_names_and_types = test_node->get_service_names_and_types();
-  EXPECT_EQ(service_names_and_types.size(), 11u);
   // These are specific to lifecycle nodes, other services are provided by rclcpp::Node
+  EXPECT_NE(
+    service_names_and_types.end(),
+    service_names_and_types.find(std::string("/testnode/change_state")));
```

Comment:
Same thing here and below; these need to be `ASSERT_NE`, otherwise we can index into the array erroneously on the next statement.

Reply:
Done